tests/test_memory.py

```diff
@@ -0,0 +1,9 @@
+from devito import DenseData
+import pytest
+import numpy as np
+
+
+@pytest.mark.parametrize('shape', [(20, 20), (20, 20, 20), (20, 20, 20, 20)])
+def test_first_touch(shape):
```

Comment:
I think a better test is to do something similar to ``test_cache_blocking_structure``, in which I search the AST for the presence of stuff like openmp pragmas etc.